dao/statistics.go

```diff
@@ -102,6 +134,45 @@ func (d statisticsDao) GetCourseStatsHourly(courseID uint) ([]Stat, error) {
 	return res, err
 }
 
+// GetCourseStatsWeekdays returns the days and their sum of vod views of a course
+func (d statisticsDao) GetLectureStatsWeekdays(courseID uint, streamID uint) ([]Stat, error) {
+	var res []Stat
+	err := DB.Raw(`SELECT DAYNAME(stats.time) AS x, SUM(stats.viewers) as y
+		FROM stats
+			JOIN streams s ON s.id = stats.stream_id
+		WHERE (s.course_id = ? OR ? = 0) AND stats.live = 0 AND stats.stream_id = ?
+		GROUP BY DAYOFWEEK(stats.time);`,
+		courseID, courseID, streamID).Scan(&res).Error
```

Comment:
`(s.course_id = ? OR ? = 0)` in `WHERE`-clause unnecessary as `stats.stream_id = ?` implicitly checks for the right course anyways?

Reply:
I ran into some errors when trying it without this check and it's also good to check if the client knows both the courseID and the streamID